source/user/df.rst

```diff
@@ -5,3 +5,143 @@ Density fitting
 
 *Modules*: :mod:`df`, :mod:`pbc.df`
 
+.. module:: df
+   :synopsis: Density fitting and RI approximation
+.. sectionauthor:: Qiming Sun <osirpt.sun@gmail.com>.
+
+Introduction
+============
+
+Density fitting (DF), sometimes also called the resolution of identity (RI) approximation, is a method to approximate the four-index electron repulsion integrals (ERIs) by two- and three-index tensors. :cite:`Dunlap00PCCP` It achieves so by expanding the atomic orbital (AO) pair density in a second, auxiliary Gaussian basis set. PySCF has extensive supports for using DF in both molecular and crystalline calculations. This section of the user document covers the basic usage of DF in molecular applications. See :ref:`theory_pbc_df` for the use of DF in periodic calculations.
+
+
+Using DF in calculations
+========================
+
+DF is not used by default when initializing a :class:`SCF` or :class:`MCSCF` object, but can be invoked via the :func:`density_fit` method as follows::
+
+    from pyscf import gto, scf, mcscf
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mf = scf.RHF(mol).density_fit().run()
+    mc = mcscf.CASSCF(mf, 8, 10).density_fit().run()
+
+Alternatively, one can initialize a :class:`DF` object and overwrite the :attr:`with_df` attribute of the :class:`SCF` object::
+
+    from pyscf import gto, scf, df
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mydf = df.DF(mol).build()
+    mf = scf.RHF(mol).density_fit()
+    mf.with_df = mydf
+    mf.kernel()
+
+Setting :attr:`with_df` to ``None`` turns DF off.
+
+Once the SCF calculations using DF are done, all post-SCF methods will use DF automatically. The example below performs a TDDFT/TDA calculation using DF::
```

Comment:
This is also weird: "once the SCF calculations using DF are done".

I think the point is that if you run a post-SCF calculation on top of a DF calculation, the post-SCF calculation will also use density fitting.

Reply:
This also leads to a question of accuracy. Let's say I run an RI-HF calculation, and then proceed with CCSD. What's the auxiliary basis used for the CCSD part? An RI-JK basis will be accurate for RI-HF, but I'm not sure if it will automatically be okay for post-HF...

Reply:
Good question. I think at least in our group when doing PBC calculations, people tend to use the same basis for mean-field and post-HF calculations. But I could be wrong and @tberkel can correct me.
I'm wondering how big a *practical* difference it will make by using e.g., a JKFIT basis for MP2 or CCSD compared to using the RI basis. @susilehtola do you know of any classical papers on this topic?

Reply:
I'm not 100% sure; there are RIMP2 auxiliary basis sets that are specifically aimed for post-HF calculations for e.g. both the Karlsruhe and the Dunning families.

Reply:
Weigend never disappoints
http://doi.org/10.1063/1.1445115

Reply:
Here's the same for Karlsruhe https://doi.org/10.1039/C4CP04286G

Reply:
Thanks! I added an example using the `cc-pVTZ-RI` basis for DF-MP2 on top of a HF calculation that does *not* use DF.